ci/ci/build.py

```diff
@@ -136,10 +137,11 @@ def __init__(self, params):
 
         self.name = json['name']
         if 'dependsOn' in json:
-            self.deps = [params.name_step[d] for d in json['dependsOn']]
+            self.deps = [params.name_step[d] for d in json['dependsOn'] if d in params.name_step]
```

Comment:
This check has to be stronger: we need to throw an error if you refer to a step that isn't defined.

Reply:
The code I had before would have done that. i.e. the dependency wasn't in name_step. Do you mean a better error message?

Reply:
I'm not sure I remember all the versions.  As written, this will silently ignore misspelled dependencies, which I don't think is OK.  master throws a KeyError, I'm fine with that.